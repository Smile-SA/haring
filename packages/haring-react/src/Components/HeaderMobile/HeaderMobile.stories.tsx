import type { Meta, StoryObj } from '@storybook/react';

import { AppShell } from '@mantine/core';
import { useStorybookArgsConnect } from '@smile/haring-react-shared/storybook-utils';
import { action } from '@storybook/addon-actions';

import {
  childrenMock,
  leftContentMock,
  rightContentMobileMock,
} from '../Header/Header.mock';

import { HeaderMobile as Cmp } from './HeaderMobile';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onSearchChange: 'searchValue',
      });
      return (
        <AppShell>
          <Story args={{ ...args }} />
        </AppShell>
      );
    },
  ],
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  title: '3-custom/Components/HeaderMobile',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const HeaderMobile: IStory = {
  args: {
    children: childrenMock(true),
    height: 90,
    left: leftContentMock,
    onSearchSubmit: action('search input submitted'),
    right: rightContentMobileMock,
    searchValue: '',
    withBorder: false,
  },
};
