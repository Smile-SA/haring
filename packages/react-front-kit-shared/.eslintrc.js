module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
