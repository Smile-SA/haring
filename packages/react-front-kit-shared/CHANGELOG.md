# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
