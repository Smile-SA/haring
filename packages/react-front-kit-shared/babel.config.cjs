const { babelConfig } = require('test');

module.exports = babelConfig;
