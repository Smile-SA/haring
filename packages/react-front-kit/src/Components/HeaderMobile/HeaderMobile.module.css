.around {
  align-items: center;
  gap: var(--mantine-spacing-xs);
}

.burgerMenu {
  background-color: var(--mantine-color-white);
  border-top: 1px solid;
  border-top-color: var(--mantine-color-gray-2);
  padding: 40px 24px;
}

.burgerSearch {
  :global(.mantine-TextInput-input) {
    padding: 6px 68px 6px 24px;
  }

  margin-top: auto;
}

.containerMobile {
  align-items: center;
  justify-content: space-between;
  padding: 16px 24px;
  position: relative;
  width: 100%;
}

.menu {
  align-items: center;
  gap: var(--mantine-spacing-xl);
  left: 50%;
  margin: auto;
  position: absolute;
  top: 50%;
  translate: -50% -50%;
}

.searchIcon {
  margin-right: 24px;
}

.separator {
  border-color: var(--mantine-color-gray-3);
}
