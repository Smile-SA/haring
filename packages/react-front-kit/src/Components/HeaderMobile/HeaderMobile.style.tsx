import { createStyles } from '@mantine/core';

export const useStyles = createStyles((theme, height: number) => ({
  around: {
    alignItems: 'center',
    gap: theme.spacing.xs,
  },
  burgerMenu: {
    backgroundColor: theme.white,
    borderTop: '1px solid',
    borderTopColor: theme.colors.gray[2],
    height: `calc(100vh - ${height}px)`,
    padding: '40px 24px',
  },
  burgerSearch: {
    '.mantine-TextInput-input': {
      padding: '6px 68px 6px 24px',
    },
    marginTop: 'auto',
  },
  containerMobile: {
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: '16px 24px',
    position: 'relative',
    width: '100%',
  },
  menu: {
    alignItems: 'center',
    gap: theme.spacing.xl,
    left: '50%',
    margin: 'auto',
    position: 'absolute',
    top: '50%',
    translate: '-50% -50%',
  },
  searchIcon: {
    marginRight: '24px',
  },
  separator: {
    borderColor: theme.colors.gray[3],
  },
}));
