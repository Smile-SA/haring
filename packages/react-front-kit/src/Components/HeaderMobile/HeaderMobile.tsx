'use client';

import type {
  BurgerProps,
  CollapseProps,
  HeaderProps,
  TextInputProps,
} from '@mantine/core';
import type { FormEvent, ReactElement, ReactNode } from 'react';

import {
  Burger,
  Collapse,
  Divider,
  Flex,
  Header as MantineHeader,
  Stack,
  TextInput,
  useMantineTheme,
} from '@mantine/core';
import { MagnifyingGlass } from '@phosphor-icons/react';
import { useState } from 'react';

import { useStyles } from './HeaderMobile.style';

export interface IHeaderMobileProps
  extends Omit<HeaderProps, 'children' | 'height' | 'left' | 'right'> {
  burgerProps?: Omit<BurgerProps, 'onClick' | 'opened'>;
  children?: ReactNode;
  collapseProps?: Omit<CollapseProps, 'in'>;
  hasSearch?: boolean;
  height?: number;
  left?: ReactNode;
  onSearchChange?: (value: string) => void;
  onSearchSubmit?: (event: FormEvent) => void;
  right?: ReactNode;
  searchInputProps?: Omit<TextInputProps, 'onChange' | 'value'>;
  searchValue?: string;
}

/** Additional props will be forwarded to the [Mantine AppShell (Header) component](https://mantine.dev/core/app-shell/) */
export function HeaderMobile(props: IHeaderMobileProps): ReactElement {
  const {
    burgerProps,
    children,
    collapseProps,
    hasSearch = true,
    height = 76,
    left,
    onSearchChange,
    onSearchSubmit,
    right,
    searchInputProps,
    searchValue,
    withBorder = true,
    ...headerProps
  } = props;
  const [burgerOpened, setBurgerOpened] = useState(false);

  const theme = useMantineTheme();
  const { classes } = useStyles(height);

  return (
    <MantineHeader height={height} withBorder={withBorder} {...headerProps}>
      <Flex className={classes.containerMobile} h={height}>
        <Flex gap={16}>
          <Burger
            color={useMantineTheme().fn.primaryColor()}
            onClick={() => setBurgerOpened((o) => !o)}
            opened={burgerOpened}
            size={22}
            title="Open navigation"
            {...burgerProps}
          />
          <Divider className={classes.separator} orientation="vertical" />
        </Flex>
        <Flex className={classes.around}>{left}</Flex>
        <Flex className={classes.around}>{right}</Flex>
      </Flex>
      <Collapse in={burgerOpened} {...collapseProps}>
        <Stack className={classes.burgerMenu}>
          {children}
          {Boolean(hasSearch) && (
            <form className={classes.burgerSearch} onSubmit={onSearchSubmit}>
              <TextInput
                aria-label="Search"
                onChange={(e) => onSearchChange?.(e.target.value)}
                placeholder="Search on the site"
                radius={32}
                rightSection={
                  <MagnifyingGlass
                    className={classes.searchIcon}
                    color={theme.fn.primaryColor()}
                    size={20}
                  />
                }
                size="md"
                value={searchValue}
                {...searchInputProps}
              />
            </form>
          )}
        </Stack>
      </Collapse>
    </MantineHeader>
  );
}
