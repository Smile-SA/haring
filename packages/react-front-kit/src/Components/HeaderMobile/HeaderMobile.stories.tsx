import type { Meta, StoryObj } from '@storybook/react';

import { useStorybookArgsConnect } from '@smile/react-front-kit-shared/storybook-utils';
import { action } from '@storybook/addon-actions';

import {
  childrenMock,
  leftContentMock,
  rightContentMobileMock,
} from '../Header/Header.mock';

import { HeaderMobile as Cmp } from './HeaderMobile';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onSearchChange: 'searchValue',
      });
      return <Story args={{ ...args }} />;
    },
  ],
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  title: '3-custom/Components/HeaderMobile',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const HeaderMobile: IStory = {
  args: {
    children: childrenMock(true),
    height: 90,
    left: leftContentMock,
    onSearchSubmit: action('search input submitted'),
    right: rightContentMobileMock,
    searchValue: '',
    withBorder: false,
  },
};
