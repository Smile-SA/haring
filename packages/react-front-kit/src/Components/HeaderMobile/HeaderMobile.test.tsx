import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { HeaderMobile } from './HeaderMobile';

describe('HeaderMobile', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <HeaderMobile searchInputProps={{ id: 'test' }}>
        <a href="#">Espace documentaire</a>
        <a href="#">Espace workflow</a>
        <a href="#">Archives</a>
      </HeaderMobile>,
    );
    expect(container).toMatchSnapshot();
  });
});
