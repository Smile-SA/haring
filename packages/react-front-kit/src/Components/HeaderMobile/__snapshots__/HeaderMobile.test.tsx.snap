// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMobile matches snapshot 1`] = `
<div>
  <header
    class="mantine-Header-root mantine-1k6wqr4"
  >
    <div
      class="mantine-bs5kw9"
    >
      <div
        class="mantine-l3a1ir"
      >
        <button
          class="mantine-UnstyledButton-root mantine-Burger-root mantine-lm35ey"
          title="Open navigation"
          type="button"
        >
          <div
            class="mantine-d9qq40 mantine-Burger-burger"
          />
        </button>
        <div
          class="mantine-Divider-root mantine-Divider-vertical mantine-1f0rfy6"
          role="separator"
        />
      </div>
      <div
        class="mantine-1dcfulj"
      />
      <div
        class="mantine-1dcfulj"
      />
    </div>
    <div
      aria-hidden="true"
      class="mantine-1avyp1d"
      style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
    >
      <div
        style="opacity: 0; transition: opacity 200ms ease;"
      >
        <div
          class="mantine-Stack-root mantine-ipqib1"
        >
          <a
            href="#"
          >
            Espace documentaire
          </a>
          <a
            href="#"
          >
            Espace workflow
          </a>
          <a
            href="#"
          >
            Archives
          </a>
          <form
            class="mantine-1cd8ioy"
          >
            <div
              class="mantine-InputWrapper-root mantine-TextInput-root mantine-q6oiko"
            >
              <div
                class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-1v7s5f8"
              >
                <input
                  aria-invalid="false"
                  aria-label="Search"
                  class="mantine-Input-input mantine-TextInput-input mantine-1d7dy6g"
                  id="test"
                  placeholder="Search on the site"
                  type="text"
                />
                <div
                  class="mantine-1swk166 mantine-Input-rightSection mantine-TextInput-rightSection"
                >
                  <svg
                    class="mantine-127k9gt"
                    fill="#0b7285"
                    height="20"
                    viewBox="0 0 256 256"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M229.66,218.34l-50.07-50.06a88.11,88.11,0,1,0-11.31,11.31l50.06,50.07a8,8,0,0,0,11.32-11.32ZM40,112a72,72,0,1,1,72,72A72.08,72.08,0,0,1,40,112Z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </header>
</div>
`;
